src/set_theory/cardinal/finite.lean

```diff
@@ -34,6 +34,64 @@ lemma card_eq_fintype_card [fintype α] : nat.card α = fintype.card α := mk_to
 @[simp]
 lemma card_eq_zero_of_infinite [infinite α] : nat.card α = 0 := mk_to_nat_of_infinite
 
+lemma card_eq_of_equiv (f : α ≃ β) : nat.card α = nat.card β :=
+begin
+  have : ulift α ≃ ulift β := (equiv.ulift.trans f).trans equiv.ulift.symm,
+  have := cardinal.to_nat_congr this,
+  simpa using this,
+end
+
+lemma card_eq_of_bijective (f : α → β) (hf : function.bijective f) : nat.card α = nat.card β :=
+card_eq_of_equiv (equiv.of_bijective f hf)
+
+lemma card_eq_of_equiv_fin {α : Type*} {n : ℕ}
+  (f : α ≃ fin n) : nat.card α = n :=
+by simpa using card_eq_of_equiv f
+
+/-- If the cardinality is positive, that means it is a finite type, so there is
+an equivalence between `α` and `fin (nat.card α)`. See also `finite.equiv_fin`. -/
+def equiv_fin_of_card_pos {α : Type*} (h : 0 < nat.card α) :
+  α ≃ fin (nat.card α) :=
+begin
+  casesI fintype_or_infinite α,
+  { simpa using fintype.equiv_fin α },
+  { simpa using h },
+end
+
+lemma card_of_subsingleton (a : α) [subsingleton α] : nat.card α = 1 :=
+begin
+  rw [card_eq_fintype_card],
+  convert fintype.card_of_subsingleton a,
+end
+
+@[simp] lemma card_unique [unique α] : nat.card α = 1 :=
+card_of_subsingleton default
+
+lemma card_eq_one_iff_nonempty_unique : nat.card α = 1 ↔ nonempty (unique α) :=
```

Comment:
We already have `cardinal.to_nat_eq_one_iff_unique`. The RHS in this lemma (and in `cardinal.eq_one_iff_unique`) is `subsingleton α ∧ nonempty α`. Could you pleae add a lemma (in `logic/unique`) that proves `subsingleton α ∧ nonempty α ↔ nonempty (unique α)`?